trollsift/parser.py

```diff
@@ -451,6 +466,45 @@ def compose(fmt, keyvals):
     return formatter.format(fmt, **keyvals)
 
 
+def partial_compose(fmt, keyvals):
+    """Convert parameters in `keyvals` to a string based on `fmt` string.
+
+    This routine is similar to compose, but accepts partial composing, i.e.,
+    not all parameters in `fmt` need to be specified in `keyvals`. Unspecified
+    parameters are left unchanged.
+
+    Args:
+        fmt (str): Python format string to match against
+        keyvals (dict): "Parameter --> parameter value" map
+
+    """
+    fmt, undefined_vars = _replace_undefined_params_with_placeholders(fmt, keyvals)
+    composed_string = compose(fmt=fmt, keyvals=keyvals)
+    for fmt_placeholder, fmt_specification in undefined_vars.items():
+        composed_string = composed_string.replace(fmt_placeholder, fmt_specification)
+
+    return composed_string
+
+
+def _replace_undefined_params_with_placeholders(fmt, keyvals=None):
+    """Replace with placeholders params in `fmt` not specified in `keyvals`."""
+    vars_left_undefined = get_convert_dict(fmt).keys()
+    if keyvals is not None:
+        vars_left_undefined -= keyvals.keys()
+
+    undefined_vars_placeholders_dict = {}
+    for var in vars_left_undefined:
+        matches = set(match.group() for match in re.finditer(rf"{{{var}([^\w{{}}].*?)*}}", fmt))
+        if len(matches) == 0:
+            raise ValueError(f"Could not capture definitions for {var} from {fmt}")
```

Comment:
Do you have any example of a call to `partial_compose` that triggers this? Could it be added as a test?

Reply:
Done